[esx]/es_extended/server/classes/player.lua

```diff
@@ -1,400 +1,425 @@
+		loadout = loadout,
+		name = name,
+		playerId = playerId,
+		source = playerId,
+		ped = GetPlayerPed(playerId),
+		variables = {},
+		weight = weight,
+		maxWeight = Config.MaxWeight,
+		license = Config.Multichar and 'license'.. identifier:sub(identifier:find(':'), identifier:len()) or 'license:'..identifier,
+
+		triggerEvent = function (self, eventName, ...)
+			TriggerClientEvent(eventName, self.playerId, ...)
+		end,
+
+		updatePed = function(self)
+			SetTimeout(1000,function()
+				if self.ped ~= GetPlayerPed(self.playerId) then
+					self.ped = GetPlayerPed(self.playerId)
+					self:triggerEvent('esx:updatePed', self.ped)
+					TriggerEvent('esx:updatePed', self.playerId, self.ped)
+					Player(self.playerId).state:set("ped", self.ped, true)
 				end
-			end
-			self.updateCoords()
-		end)
-	end
-
-	function self.getCoords(vector)
-		if vector then
-			return vector3(self.coords.x, self.coords.y, self.coords.z)
-		else
-			return self.coords
-		end
-	end
-
-	function self.kick(reason)
-		DropPlayer(self.source, reason)
-	end
-
-	function self.setMoney(money)
-		money = ESX.Math.Round(money)
-		self.setAccountMoney('money', money)
-	end
-
-	function self.getMoney()
-		return self.getAccount('money').money
-	end
-
-	function self.addMoney(money, reason)
-		money = ESX.Math.Round(money)
-		self.addAccountMoney('money', money, reason)
-	end
-
-	function self.removeMoney(money, reason)
-		money = ESX.Math.Round(money)
-		self.removeAccountMoney('money', money, reason)
-	end
-
-	function self.getIdentifier()
-		return self.identifier
-	end
-
-	function self.setGroup(newGroup)
-		ExecuteCommand(('remove_principal identifier.%s group.%s'):format(self.license, self.group))
-		self.group = newGroup
-		Player(self.source).state:set("group", self.group, true)
-		ExecuteCommand(('add_principal identifier.%s group.%s'):format(self.license, self.group))
-	end
-
-	function self.getGroup()
-		return self.group
-	end
-
-	function self.set(k, v)
-		self.variables[k] = v
-		Player(self.source).state:set(k, v, true)
-	end
+				self:updatePed()
+			end)
+		end,
+
+		getPed = function(self)
+			return self.ped
+		end,
+
+		setCoords = function(self, coord)
+			local vector = type(coord) == "vector4" and coord or type(coord) == "vector3" and vector4(coord, 0.0) or
+			vec(coord.x, coord.y, coord.z, coord.w or 0.0)
+			SetEntityCoords(self.ped, vector.x, vector.y, vector.z, false, false, false, false)
+			SetEntityHeading(self.ped, vector.w)
+		end,
+
+		updateCoords = function(self)
+			SetTimeout(1000,function()
+				if DoesEntityExist(self.ped) then
+					local coord = GetEntityCoords(self.ped)
+					local distance = #(coord - vector3(self.coords.x, self.coords.y, self.coords.z))
+					if distance > 1.5 then
+						local heading = GetEntityHeading(self.ped)
+						self.coords = {
+							x = coords.x,
+							y = coords.y,
+							z = coords.z,
+							heading = heading or 0.0
+						}
+					end
+				end
+				self:updateCoords()
+			end)
+		end,
```

Comment:
Just reminding, this loop will continue even player disconnected. I have tested it, or did I do wrong?
esx latest develop branch, server windows 6136.

Reply:
The timeout will probably run, but the update will not. You can see it here:
```lua
if DoesEntityExist(self.ped) then
```